### PR TITLE
[17.0][MIG] product_configurator

### PR DESCRIPTION
**Pull request by Nikul-OSI**

### CODE REVIEW

**Comment by Nikul-OSI** [comment]
@PCatinean @dreispt @dannyadair Can you please take a look?

**Comment by dreispt** [comment]
@Nikul-OSI Why the PATCH labels in the history commits, can you please remove them?
Also, pre-commit is failing.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -103,12 +103,6 @@ repos:
       - id: check-xml
       - id: mixed-line-ending
         args: ["--fix=lf"]
-  - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.1.3
-    hooks:
-      - id: ruff
-        args: [--fix, --exit-non-zero-on-fix]
-      - id: ruff-format
```
Comment (dreispt) [inline .pre-commit-config.yaml:111]: Why this change?

File: product_configurator/models/product_attribute.py [hunk 2]
```diff
@@ -183,16 +181,18 @@ def onchange_values(self):
     def _check_default_values(self):
         """default value should not be outside of the
         values selected in attribute line"""
-        for line in self.filtered(lambda l: l.default_val):
+        for line in self.filtered(lambda line: line.default_val):
```
Comment (dreispt) [inline product_configurator/models/product_attribute.py:184]: I'm pretty sure these changes were already in v16.
Can you check you brought over all commits there?

File: product_configurator/models/product_config.py [hunk 3]
```diff
@@ -685,7 +685,7 @@ def write(self, vals):
         try:
             self.validate_configuration(final=False)
         except ValidationError as exc:
-            raise ValidationError(_("%s") % exc.name) from exc
+            raise ValidationError(_("{}".format(exc)))
```
Comment (dreispt) [inline product_configurator/models/product_config.py:688]: Maybe `raise ValidationError(_("{}").format(exc))`?
Or even better, I can't see the point for this `except`, what if we remove it?

File: product_configurator/models/product_config.py [hunk 4]
```diff
@@ -1546,13 +1546,16 @@ def get_onchange_specifications(self, model):
...
         specs = model_obj._onchange_spec()
-        for name, field in model_obj._fields.items():
-            if field.type not in ["one2many", "many2many"]:
-                continue
-            ch_specs = self.get_child_specification(
-                model=field.comodel_name, parent=name
-            )
-            specs.update(ch_specs)
+
+        # TODO :- Commented a code and ths code already base in a odoo base modules.
```
Comment (dreispt) [inline product_configurator/models/product_config.py:1550]: Just delete the code then.

**Review by dreispt — CHANGES_REQUESTED** [review]

**Comment by dreispt** [comment]
Superseded by #117 , that has the commit history fixed.
Feel free to continue the missing work there.
